File: populate_marketplace.py
"""
Script to populate marketplace with videos from Firebase Storage
Extracts metadata, generates tags, and creates Firestore documents
"""

import os
import re
import json
import base64
from datetime import datetime, timedelta
from pathlib import Path
import firebase_admin
from firebase_admin import credentials, firestore, storage
from dotenv import load_dotenv

# Load environment variables
load_dotenv()

# Initialize Firebase Admin SDK
service_account_base64 = os.getenv('FIREBASE_SERVICE_ACCOUNT_BASE64')
if not service_account_base64:
    print('❌ FIREBASE_SERVICE_ACCOUNT_BASE64 not found in .env')
    exit(1)

service_account_json = base64.b64decode(service_account_base64).decode('utf-8')
service_account = json.loads(service_account_json)

if not firebase_admin._apps:
    cred = credentials.Certificate(service_account)
    firebase_admin.initialize_app(cred, {
        'storageBucket': os.getenv('FIREBASE_STORAGE_BUCKET', 'reelzila.firebasestorage.app')
    })

bucket = storage.bucket()
db = firestore.client()


def parse_filename(filename):
    """
    Parse video filename to extract meaningful information

    Args:
        filename (str): The video filename

    Returns:
        dict: Parsed data including title, tags, and use cases
    """
    # Remove file extension
    name_without_ext = re.sub(r'\.(mp4|mov|avi|webm)$', '', filename, flags=re.IGNORECASE)

    # Replace common separators with spaces
    clean_name = name_without_ext.replace('-', ' ').replace('_', ' ')

    # Handle camelCase
    clean_name = re.sub(r'([a-z])([A-Z])', r'\1 \2', clean_name).lower()

    # Extract words
    words = [w for w in clean_name.split() if len(w) > 2]

    # Category mapping for tags
    category_map = {
        'nature': ['nature', 'landscape', 'outdoor', 'scenery'],
        'city': ['urban', 'cityscape', 'architecture', 'downtown'],
        'ocean': ['ocean', 'sea', 'water', 'beach', 'marine'],
        'mountain': ['mountain', 'hiking', 'nature', 'landscape'],
        'sunset': ['sunset', 'golden hour', 'evening', 'sky'],
        'sunrise': ['sunrise', 'morning', 'dawn', 'sky'],
        'food': ['food', 'culinary', 'cooking', 'restaurant'],
        'people': ['people', 'lifestyle', 'social', 'human'],
        'business': ['business', 'corporate', 'professional', 'office'],
        'technology': ['technology', 'tech', 'digital', 'modern'],
        'travel': ['travel', 'vacation', 'tourism', 'destination'],
        'fitness': ['fitness', 'health', 'exercise', 'workout'],
        'abstract': ['abstract', 'artistic', 'creative', 'design'],
        'aerial': ['aerial', 'drone', 'birds eye', 'top view'],
        'slow': ['slow motion', 'slo mo', 'cinematic'],
        'time': ['time lapse', 'timelapse', 'fast motion'],
        'night': ['night', 'evening', 'dark', 'nighttime'],
        'day': ['day', 'daytime', 'bright', 'sunny'],
        'rain': ['rain', 'weather', 'wet', 'storm'],
        'snow': ['snow', 'winter', 'cold', 'white'],
        'fire': ['fire', 'flame', 'heat', 'burning'],
        'water': ['water', 'liquid', 'fluid', 'aqua'],
        'sky': ['sky', 'clouds', 'atmosphere', 'aerial'],
        'car': ['car', 'vehicle', 'automotive', 'transportation'],
        'drone': ['aerial', 'drone', 'birds eye', 'elevated view'],
        'sunset': ['golden hour', 'sunset', 'dusk', 'evening'],
        'beach': ['beach', 'coast', 'shore', 'seaside'],
        'forest': ['forest', 'woods', 'trees', 'nature'],
        'street': ['street', 'road', 'urban', 'traffic'],
        'building': ['architecture', 'building', 'structure', 'urban'],
        'cloud': ['clouds', 'sky', 'weather', 'atmosphere'],
        'woman': ['people', 'person', 'human', 'lifestyle'],
        'man': ['people', 'person', 'human', 'lifestyle'],
        'hand': ['hands', 'gestures', 'close-up', 'details'],
        'camera': ['equipment', 'filmmaking', 'photography', 'production'],
        'light': ['lighting', 'illumination', 'glow', 'ambient'],
        'wave': ['ocean', 'water', 'waves', 'coastal'],
        'tree': ['nature', 'vegetation', 'outdoor', 'landscape'],
        'road': ['road', 'highway', 'journey', 'travel'],
        'work': ['work', 'office', 'business', 'professional']
    }

    # Use case mapping
    use_case_map = {
        'nature': ['Content Creation', 'Marketing', 'Background', 'Social Media'],
        'city': ['Marketing', 'Real Estate', 'Travel', 'Background'],
        'business': ['Corporate Videos', 'Marketing', 'Presentations'],
        'food': ['Restaurant Marketing', 'Content Creation', 'Social Media'],
        'technology': ['Tech Reviews', 'Marketing', 'Presentations'],
        'abstract': ['Background', 'Transitions', 'Creative Projects'],
        'aerial': ['Real Estate', 'Travel', 'Marketing', 'Documentaries'],
        'people': ['Social Media', 'Marketing', 'Lifestyle Content'],
        'travel': ['Travel Vlogs', 'Marketing', 'Tourism', 'Documentaries']
    }

    # Generate title
    title = ' '.join(word.capitalize() for word in words)

    # Generate tags
    tags = set(['stock footage', 'video clip'])
    use_cases = set()

    for word in words:
        for category, category_tags in category_map.items():
            if category in word or word in category:
                tags.update(category_tags)
                if category in use_case_map:
                    use_cases.update(use_case_map[category])

    # Add generic use cases if none found
    if not use_cases:
        use_cases.update(['Content Creation', 'Marketing', 'Background', 'Social Media'])

    # Generate description
    description = f"High-quality {title.lower()} stock footage perfect for your creative projects. This video can be used for content creation, marketing campaigns, social media posts, and more."

    return {
        'title': title or 'Stock Video Footage',
        'tags': list(tags)[:10],  # Limit to 10 tags
        'useCases': list(use_cases)[:6],  # Limit to 6 use cases
        'description': description
    }


def get_aspect_ratio(width, height):
    """
    Determine aspect ratio from video dimensions

    Args:
        width (int): Video width
        height (int): Video height

    Returns:
        str: Aspect ratio description
    """
    if not width or not height:
        return 'Unknown'

    ratio = width / height

    if abs(ratio - 16/9) < 0.1:
        return '16:9 (Landscape)'
    elif abs(ratio - 9/16) < 0.1:
        return '9:16 (Portrait)'
    elif abs(ratio - 1) < 0.1:
        return '1:1 (Square)'
    elif abs(ratio - 4/3) < 0.1:
        return '4:3 (Standard)'
    elif abs(ratio - 21/9) < 0.1:
        return '21:9 (Ultrawide)'

    return f'{width}x{height}'


def format_duration(seconds):
    """
    Format duration from seconds to readable string

    Args:
        seconds (float): Duration in seconds

    Returns:
        str: Formatted duration
    """
    if not seconds:
        return 'Unknown'

    mins = int(seconds // 60)
    secs = int(seconds % 60)

    if mins > 0:
        return f'{mins}m {secs}s'
    return f'{secs}s'


def get_video_metadata(file_path):
    """
    Get video metadata from Firebase Storage

    Args:
        file_path (str): Path to video file in storage

    Returns:
        dict: Video metadata or None if error
    """
    try:
        blob = bucket.blob(file_path)

        # Get metadata
        blob.reload()

        # Make video publicly accessible or use a shorter-lived signed URL
        # For marketplace, we'll make the blob publicly readable
        blob.make_public()
        url = blob.public_url

        # Get custom metadata if available
        metadata_dict = blob.metadata or {}

        return {
            'url': url,
            'size': blob.size,
            'contentType': blob.content_type,
            'timeCreated': blob.time_created,
            'width': int(metadata_dict.get('width', 0)) if metadata_dict.get('width') else None,
            'height': int(metadata_dict.get('height', 0)) if metadata_dict.get('height') else None,
            'duration': float(metadata_dict.get('duration', 0)) if metadata_dict.get('duration') else None,
            'hasAudio': metadata_dict.get('hasAudio', 'false').lower() == 'true'
        }
    except Exception as e:
        print(f'  ⚠️  Error getting metadata: {str(e)}')
        return None


def create_marketplace_listing(video_data):
    """
    Create or update marketplace listing in Firestore

    Args:
        video_data (dict): Processed video data

    Returns:
        dict: Created listing or None if error
    """
    try:
        # Create new document reference
        listing_ref = db.collection('marketplace_listings').document()

        listing = {
            'id': listing_ref.id,
            'sellerId': 'admin',  # Platform curated content
            'sellerName': 'Reelzila',
            'title': video_data['title'],
            'description': video_data['description'],
            'videoUrl': video_data['url'],
            'generationId': 'imported',  # Imported from storage
            'prompt': video_data['title'].lower(),
            'price': video_data['price'],
            'tags': video_data['tags'],
            'hasAudio': video_data['hasAudio'],
            'useCases': video_data['useCases'],
            'thumbnailUrl': video_data['url'],  # Use video URL as thumbnail
            'status': 'published',
            'createdAt': firestore.SERVER_TIMESTAMP,
            'updatedAt': firestore.SERVER_TIMESTAMP,
            'sold': 0,
            # Additional metadata
            'aspectRatio': video_data['aspectRatio'],
            'duration': video_data['duration'],
            'durationSeconds': video_data['durationSeconds'],
            'resolution': video_data['resolution'],
            'fileSize': video_data['fileSize'],
            'fileName': video_data['fileName']
        }

        listing_ref.set(listing)
        print(f'  ✓ Created listing: {video_data["title"]} ({video_data["fileName"]})')

        return listing
    except Exception as e:
        print(f'  ❌ Error creating listing: {str(e)}')
        return None


def populate_marketplace():
    """
    Main function to process all videos
    """
    print('🚀 Starting marketplace population...\n')

    try:
        # List all blobs in storage
        blobs = list(bucket.list_blobs())

        # Filter video files
        video_extensions = ['.mp4', '.mov', '.avi', '.webm']
        video_blobs = [
            blob for blob in blobs
            if any(blob.name.lower().endswith(ext) for ext in video_extensions)
        ]

        print(f'📹 Found {len(video_blobs)} video files in storage\n')

        if len(video_blobs) == 0:
            print('No video files found. Make sure videos are uploaded to Firebase Storage.')
            return

        # Process each video
        success_count = 0
        error_count = 0

        for i, blob in enumerate(video_blobs):
            file_name = Path(blob.name).name

            print(f'\n[{i + 1}/{len(video_blobs)}] Processing: {file_name}')

            # Get metadata from storage
            metadata = get_video_metadata(blob.name)

            if not metadata:
                print(f'  ⚠️  Skipping - could not get metadata')
                error_count += 1
                continue

            # Parse filename
            parsed = parse_filename(file_name)

            # Determine pricing based on characteristics
            price = 4.99  # Default
            if metadata['duration'] and metadata['duration'] > 30:
                price = 7.99  # Longer videos
            if metadata['width'] and metadata['width'] >= 3840:
                price = 9.99  # 4K videos

            # Combine all data
            video_data = {
                'fileName': file_name,
                'url': metadata['url'],
                'title': parsed['title'],
                'description': parsed['description'],
                'tags': parsed['tags'],
                'useCases': parsed['useCases'],
                'hasAudio': metadata['hasAudio'],
                'aspectRatio': get_aspect_ratio(metadata['width'], metadata['height']),
                'duration': format_duration(metadata['duration']),
                'durationSeconds': metadata['duration'],
                'resolution': f"{metadata['width']}x{metadata['height']}" if metadata['width'] and metadata['height'] else 'Unknown',
                'fileSize': f"{(metadata['size'] / (1024 * 1024)):.2f} MB",
                'price': price
            }

            # Log extracted info
            print(f'  📝 Title: {video_data["title"]}')
            print(f'  🏷️  Tags: {", ".join(video_data["tags"])}')
            print(f'  📊 Aspect Ratio: {video_data["aspectRatio"]}')
            print(f'  ⏱️  Duration: {video_data["duration"]}')
            print(f'  📐 Resolution: {video_data["resolution"]}')
            print(f'  🔊 Audio: {"Yes" if video_data["hasAudio"] else "No"}')
            print(f'  💰 Price: €{video_data["price"]}')

            # Create listing
            listing = create_marketplace_listing(video_data)

            if listing:
                success_count += 1
            else:
                error_count += 1

        print('\n' + '=' * 60)
        print('✅ Marketplace population complete!')
        print(f'  - Successfully processed: {success_count} videos')
        print(f'  - Errors: {error_count}')
        print(f'  - Total: {len(video_blobs)} videos')
        print('=' * 60 + '\n')

    except Exception as e:
        print(f'❌ Fatal error: {str(e)}')
        import traceback
        traceback.print_exc()
        exit(1)


if __name__ == '__main__':
    populate_marketplace()
    print('Script completed successfully!')

File: populate-marketplace.js
/**
 * Script to populate marketplace with videos from Firebase Storage
 * Extracts metadata, generates tags, and creates Firestore documents
 */

const admin = require('firebase-admin');
const path = require('path');
const fs = require('fs');

// Load environment variables
require('dotenv').config();

// Initialize Firebase Admin SDK
const serviceAccountBase64 = process.env.FIREBASE_SERVICE_ACCOUNT_BASE64;
if (!serviceAccountBase64) {
  console.error('FIREBASE_SERVICE_ACCOUNT_BASE64 not found in .env');
  process.exit(1);
}

const serviceAccountJson = Buffer.from(serviceAccountBase64, 'base64').toString('utf-8');
const serviceAccount = JSON.parse(serviceAccountJson);

admin.initializeApp({
  credential: admin.credential.cert(serviceAccount),
  storageBucket: process.env.FIREBASE_STORAGE_BUCKET || 'reelzila.firebasestorage.app'
});

const bucket = admin.storage().bucket();
const db = admin.firestore();

/**
 * Parse video filename to extract meaningful information
 * @param {string} filename - The video filename
 * @returns {object} Parsed data including title, tags, and use cases
 */
function parseFilename(filename) {
  // Remove file extension and replace separators with spaces
  const nameWithoutExt = filename.replace(/\.(mp4|mov|avi|webm)$/i, '');

  // Replace common separators with spaces
  let cleanName = nameWithoutExt
    .replace(/[-_]/g, ' ')
    .replace(/([a-z])([A-Z])/g, '$1 $2') // Handle camelCase
    .toLowerCase();

  // Extract potential keywords for tags
  const words = cleanName.split(/\s+/).filter(w => w.length > 2);

  // Common video categories and their associated tags
  const categoryMap = {
    'nature': ['nature', 'landscape', 'outdoor', 'scenery'],
    'city': ['urban', 'cityscape', 'architecture', 'downtown'],
    'ocean': ['ocean', 'sea', 'water', 'beach', 'marine'],
    'mountain': ['mountain', 'hiking', 'nature', 'landscape'],
    'sunset': ['sunset', 'golden hour', 'evening', 'sky'],
    'sunrise': ['sunrise', 'morning', 'dawn', 'sky'],
    'food': ['food', 'culinary', 'cooking', 'restaurant'],
    'people': ['people', 'lifestyle', 'social', 'human'],
    'business': ['business', 'corporate', 'professional', 'office'],
    'technology': ['technology', 'tech', 'digital', 'modern'],
    'travel': ['travel', 'vacation', 'tourism', 'destination'],
    'fitness': ['fitness', 'health', 'exercise', 'workout'],
    'abstract': ['abstract', 'artistic', 'creative', 'design'],
    'aerial': ['aerial', 'drone', 'birds eye', 'top view'],
    'slow motion': ['slow motion', 'slo mo', 'cinematic'],
    'time lapse': ['time lapse', 'timelapse', 'fast motion'],
    'night': ['night', 'evening', 'dark', 'nighttime'],
    'day': ['day', 'daytime', 'bright', 'sunny'],
    'rain': ['rain', 'weather', 'wet', 'storm'],
    'snow': ['snow', 'winter', 'cold', 'white'],
    'fire': ['fire', 'flame', 'heat', 'burning'],
    'water': ['water', 'liquid', 'fluid', 'aqua'],
    'sky': ['sky', 'clouds', 'atmosphere', 'aerial']
  };

  // Use case mapping based on content
  const useCaseMap = {
    'nature': ['Content Creation', 'Marketing', 'Background', 'Social Media'],
    'city': ['Marketing', 'Real Estate', 'Travel', 'Background'],
    'business': ['Corporate Videos', 'Marketing', 'Presentations'],
    'food': ['Restaurant Marketing', 'Content Creation', 'Social Media'],
    'technology': ['Tech Reviews', 'Marketing', 'Presentations'],
    'abstract': ['Background', 'Transitions', 'Creative Projects'],
    'aerial': ['Real Estate', 'Travel', 'Marketing', 'Documentaries']
  };

  // Generate title (capitalize first letter of each word)
  const title = words
    .map(w => w.charAt(0).toUpperCase() + w.slice(1))
    .join(' ');

  // Generate tags based on keywords found
  const tags = new Set(['stock footage', 'video clip']);
  const useCases = new Set();

  words.forEach(word => {
    // Check if word matches any category
    Object.keys(categoryMap).forEach(category => {
      if (word.includes(category) || category.includes(word)) {
        categoryMap[category].forEach(tag => tags.add(tag));
        if (useCaseMap[category]) {
          useCaseMap[category].forEach(uc => useCases.add(uc));
        }
      }
    });
  });

  // Add generic use cases if none found
  if (useCases.size === 0) {
    ['Content Creation', 'Marketing', 'Background', 'Social Media'].forEach(uc => useCases.add(uc));
  }

  // Generate description
  const description = `High-quality ${title.toLowerCase()} stock footage perfect for your creative projects. This video can be used for content creation, marketing campaigns, social media posts, and more.`;

  return {
    title: title || 'Stock Video Footage',
    tags: Array.from(tags).slice(0, 10), // Limit to 10 tags
    useCases: Array.from(useCases).slice(0, 6), // Limit to 6 use cases
    description
  };
}

/**
 * Determine aspect ratio from video dimensions
 * @param {number} width - Video width
 * @param {number} height - Video height
 * @returns {string} Aspect ratio description
 */
function getAspectRatio(width, height) {
  if (!width || !height) return 'Unknown';

  const ratio = width / height;

  if (Math.abs(ratio - 16/9) < 0.1) return '16:9 (Landscape)';
  if (Math.abs(ratio - 9/16) < 0.1) return '9:16 (Portrait)';
  if (Math.abs(ratio - 1) < 0.1) return '1:1 (Square)';
  if (Math.abs(ratio - 4/3) < 0.1) return '4:3 (Standard)';
  if (Math.abs(ratio - 21/9) < 0.1) return '21:9 (Ultrawide)';

  return `${width}x${height}`;
}

/**
 * Format duration from seconds to readable string
 * @param {number} seconds - Duration in seconds
 * @returns {string} Formatted duration
 */
function formatDuration(seconds) {
  if (!seconds) return 'Unknown';

  const mins = Math.floor(seconds / 60);
  const secs = Math.floor(seconds % 60);

  if (mins > 0) {
    return `${mins}m ${secs}s`;
  }
  return `${secs}s`;
}

/**
 * Get video metadata from Firebase Storage
 * @param {string} filePath - Path to video file in storage
 * @returns {object} Video metadata
 */
async function getVideoMetadata(filePath) {
  try {
    const file = bucket.file(filePath);
    const [metadata] = await file.getMetadata();

    // Get public URL
    const [url] = await file.getSignedUrl({
      action: 'read',
      expires: '03-01-2500' // Far future date for marketplace videos
    });

    // Extract custom metadata if available
    const customMetadata = metadata.metadata || {};

    return {
      url,
      size: metadata.size,
      contentType: metadata.contentType,
      timeCreated: metadata.timeCreated,
      width: customMetadata.width ? parseInt(customMetadata.width) : null,
      height: customMetadata.height ? parseInt(customMetadata.height) : null,
      duration: customMetadata.duration ? parseFloat(customMetadata.duration) : null,
      hasAudio: customMetadata.hasAudio === 'true' || customMetadata.hasAudio === true
    };
  } catch (error) {
    console.error(`Error getting metadata for ${filePath}:`, error.message);
    return null;
  }
}

/**
 * Create or update marketplace listing in Firestore
 * @param {object} videoData - Processed video data
 */
async function createMarketplaceListing(videoData) {
  try {
    const listingRef = db.collection('marketplace_listings').doc();

    const listing = {
      id: listingRef.id,
      sellerId: 'admin', // Platform curated content
      sellerName: 'Reelzila',
      title: videoData.title,
      description: videoData.description,
      videoUrl: videoData.url,
      generationId: 'imported', // Imported from storage
      prompt: videoData.title.toLowerCase(), // Use title as prompt
      price: videoData.price || 4.99, // Default price €4.99
      tags: videoData.tags,
      hasAudio: videoData.hasAudio,
      useCases: videoData.useCases,
      thumbnailUrl: videoData.url, // Use video URL as thumbnail (can be improved)
      status: 'published',
      createdAt: admin.firestore.FieldValue.serverTimestamp(),
      updatedAt: admin.firestore.FieldValue.serverTimestamp(),
      sold: 0,
      // Additional metadata for search/filter
      aspectRatio: videoData.aspectRatio,
      duration: videoData.duration,
      durationSeconds: videoData.durationSeconds,
      resolution: videoData.resolution,
      fileSize: videoData.fileSize,
      fileName: videoData.fileName
    };

    await listingRef.set(listing);
    console.log(`✓ Created listing: ${videoData.title} (${videoData.fileName})`);

    return listing;
  } catch (error) {
    console.error(`Error creating listing for ${videoData.fileName}:`, error.message);
    return null;
  }
}

/**
 * Main function to process all videos
 */
async function populateMarketplace() {
  console.log('🚀 Starting marketplace population...\n');

  try {
    // List all video files in storage
    const [files] = await bucket.getFiles();

    // Filter video files
    const videoFiles = files.filter(file => {
      const ext = path.extname(file.name).toLowerCase();
      return ['.mp4', '.mov', '.avi', '.webm'].includes(ext);
    });

    console.log(`📹 Found ${videoFiles.length} video files in storage\n`);

    if (videoFiles.length === 0) {
      console.log('No video files found. Make sure videos are uploaded to Firebase Storage.');
      return;
    }

    // Process each video
    let successCount = 0;
    let errorCount = 0;

    for (let i = 0; i < videoFiles.length; i++) {
      const file = videoFiles[i];
      const fileName = path.basename(file.name);

      console.log(`\n[${i + 1}/${videoFiles.length}] Processing: ${fileName}`);

      // Get metadata from storage
      const metadata = await getVideoMetadata(file.name);

      if (!metadata) {
        console.log(`  ⚠️  Skipping - could not get metadata`);
        errorCount++;
        continue;
      }

      // Parse filename to extract information
      const parsed = parseFilename(fileName);

      // Determine pricing based on video characteristics
      let price = 4.99; // Default
      if (metadata.duration && metadata.duration > 30) price = 7.99; // Longer videos
      if (metadata.width && metadata.width >= 3840) price = 9.99; // 4K videos

      // Combine all data
      const videoData = {
        fileName,
        url: metadata.url,
        title: parsed.title,
        description: parsed.description,
        tags: parsed.tags,
        useCases: parsed.useCases,
        hasAudio: metadata.hasAudio,
        aspectRatio: getAspectRatio(metadata.width, metadata.height),
        duration: formatDuration(metadata.duration),
        durationSeconds: metadata.duration,
        resolution: metadata.width && metadata.height ? `${metadata.width}x${metadata.height}` : 'Unknown',
        fileSize: `${(metadata.size / (1024 * 1024)).toFixed(2)} MB`,
        price
      };

      // Log extracted info
      console.log(`  📝 Title: ${videoData.title}`);
      console.log(`  🏷️  Tags: ${videoData.tags.join(', ')}`);
      console.log(`  📊 Aspect Ratio: ${videoData.aspectRatio}`);
      console.log(`  ⏱️  Duration: ${videoData.duration}`);
      console.log(`  📐 Resolution: ${videoData.resolution}`);
      console.log(`  🔊 Audio: ${videoData.hasAudio ? 'Yes' : 'No'}`);
      console.log(`  💰 Price: €${videoData.price}`);

      // Create listing in Firestore
      const listing = await createMarketplaceListing(videoData);

      if (listing) {
        successCount++;
      } else {
        errorCount++;
      }

      // Small delay to avoid rate limiting
      await new Promise(resolve => setTimeout(resolve, 500));
    }

    console.log('\n' + '='.repeat(60));
    console.log('✅ Marketplace population complete!');
    console.log(`  - Successfully processed: ${successCount} videos`);
    console.log(`  - Errors: ${errorCount}`);
    console.log(`  - Total: ${videoFiles.length} videos`);
    console.log('='.repeat(60) + '\n');

  } catch (error) {
    console.error('❌ Fatal error:', error);
    process.exit(1);
  }
}

// Run the script
populateMarketplace()
  .then(() => {
    console.log('Script completed successfully!');
    process.exit(0);
  })
  .catch(error => {
    console.error('Script failed:', error);
    process.exit(1);
  });
